src/imagery/i.landsat/i.landsat.download/i.landsat.download.py

```diff
@@ -206,108 +492,74 @@ def main():
-                print(
-                    scene["entity_id"],
-                    scene["display_id"],
-                    scene["acquisition_date"].strftime("%Y-%m-%d"),
-                    scene["cloud_cover"],
-                )
-
-            gs.message(
-                _(
-                    "To download all scenes found, re-run the previous "
-                    "command without -l flag. Note that if no output "
-                    "option is provided, files will be downloaded in /tmp"
-                )
+        # Should use other provider other than USGS,
+        # as there was a bug in USGS API, fixed here
+        # https://github.com/CS-SI/eodag/issues/1252
+        # Currently availble in EODAG 3.0.0b3 (not yet released)
+        downloader.set_scenes_ids(options["id"].split(","))
+        # TODO: Remove when deserialize_and_register bug is fix
+        if options["datasource"] != "usgs":
+            downloader.download(outdir)
+        return 0
+
+    # Translate sort option to the i.eodag sort options
+    eodag_sort = ""
+    for sort_var in options["sort"].split(","):
+        if sort_var == "cloud_cover":
+            eodag_sort += "cloudcover,"
+        if sort_var == "acquisition_date":
+            eodag_sort += "ingestiondate,"
+
+    # Searching is necessary before listing or downloading
+    downloader.search(
+        dataset=options["dataset"],
+        start=start_date,
+        end=end_date,
+        clouds=options["clouds"] if options["clouds"] else None,
+        vector_map=options["map"] if options["map"] else None,
+        limit=options["limit"],
+        eodag_sort=eodag_sort,
+        eodag_order=options["order"],
+    )
+
+    if flags["l"]:
+        downloader.list()
+        gs.message(
+            _(
+                "To download all scenes found, re-run the previous "
+                "command without -l flag. Note that if no output "
+                "option is provided, files will be downloaded in /tmp"
             )
-
-        else:
-
-            ee = EarthExplorer(user, password)
-
-            for scene in sorted_scenes:
-
-                gs.message(_("Downloading scene <{}> ...").format(scene["entity_id"]))
-
-                ee.download(
-                    identifier=scene["entity_id"],
-                    output_dir=outdir,
-                    timeout=int(options["timeout"]),
-                )
-
-            ee.logout()
+        )
+    else:
+        downloader.download(outdir)
 
 
 if __name__ == "__main__":
     options, flags = gs.parser()
-
-    # lazy import
     try:
-        import landsatxplore.api
-        from landsatxplore.earthexplorer import EarthExplorer
-
-    except ImportError:
-
-        gs.fatal(_("Cannot import landsatxplore. Please install the library first."))
-
-    main()
+        import eodag
+        from eodag import EODataAccessGateway
+    except:
+        gs.fatal(_("Cannot import eodag. Please intall the library first."))
+
```

Comment:
You should check if _i.eodag_ is installed with e.g.: [find_program](https://grass.osgeo.org/grass83/manuals/libpython/grass.script.html?highlight=find_program#grass.script.core.find_program)

```suggestion
try:
    gs.find_program("i.eodag", "--help")
except ImportError:
    gs.fatal(_("Addon i.eodag not found. Please intall it with g.extension."))
```
Alternatively, if you use `Module()` in _main()_, you could do something along the lines of:

```
try:
    i_eodag = Module("i.eodag", run_=False)
except ImportError:
    gs.fatal(_("Addon i.eodag not found. Please intall it with g.extension."))
```